library/Vanilla/Formatting/Html/DomUtils.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @author Dani M. <dani,m@vanillaforums.com>
+ * @author Isis Graziatto <isis.g@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting\Html;
+
+use DOMDocument;
+
+/**
+ * Class for stripping images and truncating text from a Dom.
+ */
+final class DomUtils {
+
+    /** @var array */
+    const EMBED_CLASSES = ['js-embed', 'embedResponsive', 'embedExternal', 'embedImage', 'VideoWrap', 'iframe'];
+
+    /**
+     * Remove embeds from the dom.
+     *
+     * @param DOMDocument $dom
+     * @param array $embedClasses
+     */
+    public static function stripEmbeds(DOMDocument $dom, array $embedClasses = self::EMBED_CLASSES): void {
+        $xpath = new \DomXPath($dom);
+        foreach ($embedClasses as $key => $value) {
+            $xpathQuery = $xpath->query(".//*[contains(@class, '$embedClasses[$key]')]");
+            $xpathDivQuery = $xpath->query("//div[@data-embedjson]");
+            $dataClassItem = $xpathQuery->item(0);
+            $dataDivItem = $xpathDivQuery->item(0);
+            if ($dataClassItem) {
+                $dataClassItem->parentNode->removeChild($dataClassItem);
+            } elseif ($dataDivItem) {
+                $dataDivItem->parentNode->removeChild($dataDivItem);
+            }
+        }
+    }
+
+    /**
+     * Remove images from the dom.
+     *
+     * @param DOMDocument $dom
+     */
+    public static function stripImages(DOMDocument $dom): void {
+        $domImages = $dom->getElementsByTagName('img');
+        $imagesArray = [];
+        foreach ($domImages as $domImage) {
+            $imagesArray[] = $domImage;
+        }
+        foreach ($imagesArray as $domImage) {
+            $domImage->parentNode->removeChild($domImage);
+        }
+    }
+
+    /**
+     * Prepare the html string that will be truncated.
+     *
+     * @param DOMDocument $dom
+     * @param int $wordCount Number of words to truncate to
+     */
+    public static function trimWords(DOMDocument $dom, int $wordCount): void {
+        $wordCounter = $wordCount;
+        self::truncateWordsRecursive($dom->documentElement, $wordCounter, $wordCount);
+    }
+
+    /**
+     * Recursively truncate text while preserving html tags.
+     *
+     * @param mixed $element Dom element.
+     * @param int $wordCounter Counter of number of word to trucnate to.
+     * @param int $wordCount Number of words to truncate to.
+     * @return int Return limit used to count remaining tags.
+     */
+    private static function truncateWordsRecursive($element, int $wordCounter, int $wordCount): int {
+        if ($wordCounter > 0) {
+            // Nodetype text
+            if ($element->nodeType == 3) {
```

Comment:
Avoid magic numbers when possible. It looks like there are [constants](https://www.php.net/manual/en/dom.constants.php) for these.